aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -278,7 +278,8 @@ public void initialize(Map<String, String> properties) {
       this.metrics = ctor.newInstance("s3");
 
       metrics.initialize(properties);
-    } catch (NoSuchMethodException | ClassCastException e) {
+    } catch (NoClassDefFoundError | NoSuchMethodException | ClassCastException e) {
+      this.metrics = MetricsContext.nullMetrics();
```

Comment:
[question] I believe we have set metrics to nullMetrics already here : 
https://github.com/apache/iceberg/blob/7c2ea0128133f630041ecb77de1fae754073e904/aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java#L76

I belive then this would not be required here ? Should we do the same for all remaining filesIO's as well just in case someone tries to access the FileIO obj , without initialization (not sure how practical this case is though), Your thoughts

Reply:
I don't see any harm in initialising at a class level (might be helpful for some use cases, maybe a test?). I can be convinced otherwise though.

The scenario here is - during runtime metrics is overridden at `this.metrics = ctor.newInstance("scheme");` and then the exception happens in `metrics.initialize(properties);` leading to no fallback.